### PR TITLE
SA-417: Fixing a whole heap of minor annyoances

### PR DESCRIPTION
**Pull request by atomfinger**
Here's what this PR address:

- Change image to openjdk:16-buster and use JDK `temurin`. At some point, we might want to have a conversation about which JDK we want to use, but temurin is legit enough that I'm not really worried.
- Remove minor versions from images
- Remove double space in copy in the dockerfile
- Remove volume in the dockerfile
- Remove apmVersion dependency variable - let dependabot deal with it
- Add annotation processor for spring boot configuration processor
- Update group name in gradle files (also make sure this updates in the python script)
- Remove WebConfig with application.yml servlet.context-path: /api
- Main: TimeZone.setDefault(TimeZone.getTimeZone(ZoneOffset.UTC))
- Template for operations (see bob-integration readme)
- Application.yml: logging: config: classpath etc (see bob integration)
- Add SonarCloud implementation and update documentation on how to enable it.

SonarCloud documentation can be found [here](https://github.com/protectorinsurance/protector-initializr-java/wiki/Getting-started#step-6-add-sonarcloud). I will add a few links and so when this gets merged to main.

### CODE REVIEW

File: kafka/src/main/resources/log-config-system-test.xml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<configuration debug="true">
```
Comment (atomfinger) [inline kafka/src/main/resources/log-config-system-test.xml:1]: This config is just to make it easier to read the logs from the system tests

File: README.md [hunk 2]
```diff
@@ -23,6 +23,25 @@ _Note: systems tests do not execute without the `systemTest` parameter. This is
 
 Go to Application and run the main method. Intellij should pick it up.
 
+## Operational information
+
+__[TODO]__: Once the system has been put in to production, make sure to
+include information about where to reach teh service and other relevant
```
Comment (pfs-olwe99) [inline README.md:29]: Typo: "teh"

File: kafka/src/main/resources/log-config-system-test.xml (hunk 1 above)
Comment (pfs-olwe99) [reply kafka/src/main/resources/log-config-system-test.xml:1]: Does it need to be in the kafka subproject? There is also one in the web subproject.

Comment (atomfinger) [reply kafka/src/main/resources/log-config-system-test.xml:1]: Both are being run as system tests

File: web/src/main/resources/log-config-system-test.xml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<configuration debug="true">
```
Comment (pfs-olwe99) [inline web/src/main/resources/log-config-system-test.xml:1]: Remember to reference this file with logging.config in application-system-test.yml

**Review by pfs-olwe99 — APPROVED** [review]

**Pull request merged**